marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flow/testTargetCollectionAdditivity/flowWithTargetCollectionsAdditivity.sjs

```diff
@@ -0,0 +1,74 @@
+const DataHubSingleton = require("/data-hub/5/datahub-singleton.sjs");
+const datahub = DataHubSingleton.instance();
+const test = require("/test/test-helper.xqy");
+
+let assertions = [];
+let results;
+let output_collection, source_collection;
+
+var options = {
+  sourceDatabase: "data-hub-FINAL",
+  targetDatabase: "data-hub-FINAL",
+  targetCollectionsAdditivity: false,
+  outputFormat: 'json'
+};
+/*
+const content = ["/customer1.json"].map(uri => {
+  return {
+    uri: uri,
+    value: cts.doc(uri)
+  };
+});
+*/
+const content = datahub.flow.findMatchingContent('simpleMappingFlow', 1, options);
+
+xdmp.log("content: " + JSON.stringify(content));
+
+results = datahub.flow.runFlow('simpleMappingFlow', 'test-job', content, options, 1);
+
+// collection specified by step definition
+// output document should have this collection regardless of settings
+output_collection = getUris("Customer");
+// collection inherited from source data
+// output document should only have this when targetCollectionsAdditivity is true or "true"
+source_collection = getUris("customer-input");
+
+assertions.push(
+  // one document is written
+  test.assertEqual(1, results.totalCount),
+  test.assertEqual(0, results.errorCount),
+  test.assertEqual(1, results.completedItems.length),
+
+  // document should have "Customer" collection as specified by artifact
+  test.assertEqual(1, output_collection.length),
+  test.assertTrue(output_collection[0] == "/customer1.json"),
+
+  // document should not inherit "customer-input" collection from ingested data
+  test.assertEqual(0, source_collection.length)
+);
+
+options.targetCollectionsAdditivity = true;
+
+results = datahub.flow.runFlow('simpleMappingFlow', 'test-job', content, options, 1);
+output_collection = getUris("Customer");
+source_collection = getUris("customer-input");
+
+assertions.push(
+  // one document is written
+  test.assertEqual(1, results.totalCount),
+  test.assertEqual(0, results.errorCount),
+  test.assertEqual(1, results.completedItems.length),
+  
+  // document should have "Customer" collection as before
+  test.assertEqual(1, output_collection.length),
+
+  // document should have "customer-input" collection preserved by targetCollectionsAddivity
+  test.assertEqual(1, source_collection.length),
+  test.assertTrue(source_collection[0] == "/customer1.json")
+);
+
+function getUris(collection){
```

Comment:
No need for this - use hubTest.getRecord and then verify that the collections on the record are correct - i.e. that the collections are either retained or not retained.